core/src/runtime/mod.rs

```diff
@@ -1255,6 +1287,73 @@ impl<'a> Runtime<'a> {
     }
 }
 
+pub trait LookupIdSampler {
```

Comment:
why is this a trait? when would another struct implement this

Reply:
Main reason is that seems weird to pass in a reference to a runtime object here: https://github.com/succinctlabs/sp1/blob/f2ece44c6f303e8b785c958ff62d90ddf185af8d/core/src/cpu/event.rs#L75